File: SELL3A Castomer App/Android/Souq-development/app/src/main/java/ahmed/adel/sleeem/clowyy/souq/pojo/response/ProductResponse.kt
package ahmed.adel.sleeem.clowyy.souq.pojo.response

import java.io.Serializable

class ProductResponse : ArrayList<ProductResponse.Item>(){


    data class Item(
        val brand: String,
        val category: Category,
        val color: List<String>,
        val companyName: String,
        val description: String,
        val id: Int,
        val image: String,
        val price: Double,
        var quantity: Int,
        val rating: Double,
        val sale: Sale? = null,
        val size: List<String>,
        val title: String,

        //cart data
        var selectedColor : String ? = null,
        var selectedSize : String ? = null,
        var totalPrice : String ? = null,
        var countOfSelectedItem : Int  = 1

    ) : Serializable{
        data class Category(
            val name: String,
            val url: String
        )
        data class Sale(
            val amount: Int,
            val duration: String,
            val image: List<String>?,
            val type: String
        )
    }
}

File: SELL3A Customer App/IOS/Sell3a/Model/Response/ProductResponse.swift
//
//  ProductResponse.swift
//  Sel3a
//
//  Created by Mahmoud Mousa on 18/06/2021.
//

import Foundation
struct ProductResponse: Codable {
       let color, size: [String]?
       let brand: String
       let category: Category
       let companyName:String
       let description: String
       let id: Int
       let image: String
       let price: Double
       let quantity: Int
       let sale: Sale?
       let title: String
       let v: Int?
       let rating: Double?
   }

   // MARK: - Category
   struct Category: Codable {
       let name: String
       let url: String
   }

   // MARK: - Sale
   struct Sale: Codable {
       let amount: Int
       let duration: String
       let image: [String]?
       let type: String
   }
